gramps/plugins/lib/libgedcom.py

```diff
@@ -1248,41 +1248,37 @@ def report_error(self, problem, line):
 
 class UTF8Reader(BaseReader):
 
-    def __init__(self, ifile, __add_msg):
-        BaseReader.__init__(self, ifile, 'utf8', __add_msg)
+    def __init__(self, ifile, __add_msg, enc):
+        BaseReader.__init__(self, ifile, enc, __add_msg)
         self.reset()
-
-    def reset(self):
-        self.ifile.seek(0)
-        data = self.ifile.read(3)
-        if data != b"\xef\xbb\xbf":
-            self.ifile.seek(0)
+        if enc == 'UTF_8_SIG':
+            self.ifile = TextIOWrapper(ifile, encoding='utf_8_sig', errors='replace', newline=None)
```

Comment:
Not sure why you can't just pass `enc` directly to `TextIOWrapper`. Note that encodings are case-insensitive.